### PR TITLE
Update the build and packaging script for Fedora 37.

### PR DESCRIPTION
**Pull request by jewelpit**
Since Ubuntu dropped support for building RPMs in 20.04, but we need to build against a version of GNOME that didn't ship until Ubuntu 22.10, we have to split the build into a `.deb` half and a `.rpm` half. The `.deb` half must be run under Debian or Ubuntu, while the `.rpm` half must be run on a Fedora or RHEL machine. This allows us to release coreyberla's fix, resolving #108 and #101.

New package metadata (package name and signature match):
```
[jpitts@fedora nautilus-dropbox]$ rpm -qip /tmp/dbx_build/result/fedora/37/x86_64/nautilus-dropbox-2022.12.05-1.fedora.x86_64.rpm 
Name        : nautilus-dropbox
Version     : 2022.12.05
Release     : 1.fc37
Architecture: x86_64
Install Date: (not installed)
Group       : User Interface/Desktops
Size        : 205877
License     : GPLv3
Signature   : RSA/SHA256, Wed 07 Dec 2022 11:17:11 AM PST, Key ID fc918b335044912e
Source RPM  : nautilus-dropbox-2022.12.05-1.fc37.src.rpm
Build Date  : Wed 07 Dec 2022 11:16:46 AM PST
Build Host  : fedora.local
URL         : https://www.dropbox.com/
Summary     : Dropbox integration for Nautilus
Description :
Nautilus Dropbox is an extension that integrates
the Dropbox web service with your GNOME Desktop.

Check us out at https://www.dropbox.com/
```

Old package metadata:
```
[jpitts@fedora nautilus-dropbox]$ rpm -qip 35/x86_64/nautilus-dropbox-2020.03.04-1.fedora.x86_64.rpm 
Name        : nautilus-dropbox
Version     : 2020.03.04
Release     : 1.fc21
Architecture: x86_64
Install Date: (not installed)
Group       : User Interface/Desktops
Size        : 198248
License     : GPLv3
Signature   : RSA/SHA1, Wed 04 Mar 2020 03:33:07 PM PST, Key ID fc918b335044912e
Source RPM  : nautilus-dropbox-2020.03.04-1.fc21.src.rpm
Build Date  : Wed 04 Mar 2020 03:33:05 PM PST
Build Host  : vagrant.vm
URL         : https://www.dropbox.com/
Summary     : Dropbox integration for Nautilus
Description :
Nautilus Dropbox is an extension that integrates
the Dropbox web service with your GNOME Desktop.

Check us out at https://www.dropbox.com/
```

### CODE REVIEW

**Comment by jewelpit** [comment]
This is now out of WIP and ready for review :)

**Comment by jewelpit** [comment]
Okay *now* it's ready.

**Review by goffrie — APPROVED** [review]
If it works it works, I admit that I don't really know how rpm works.

Incidentally we should probably also get rid of the sketchy code included in the RPM that traverses /home/*/.dropbox-dist and deletes old Dropbox versions (see generate-rpm.sh)

File: HOWTO_PACKAGE.md [hunk 1]
```diff
@@ -2,46 +2,60 @@
...
-sudo DIST=jessie ARCH=amd64 pbuilder create --debootstrapopts --variant=buildd
-```
+   4. Build all the packages with: `python3 build_packages.py deb`
 
-5. Add yourself to the mock group (create it if it doesn't exist), and
-   then logout and login for the group to take affect.
-   Probably you have to run `build_packages.py` command with `sudo` permission which 
-   means you should add your root account to the mock group. Otherwise, you will see
-   `The password you typed is invalid.` error.
+4. On a Fedora machine:
```
[the remark this answers is not in the document]
Comment (jewelpit) [reply HOWTO_PACKAGE.md:31]: Step three knows what it did >:(

File: generate-rpm.sh [hunk 2]
```diff
@@ -227,29 +218,22 @@ fi
...
+# Old versions of the rpm delete these files in postun.  Fortunately we have saved a backup.
+if [ ! -e %{_libdir}/nautilus/extensions-4/libnautilus-dropbox.so ]; then
+  if [ -e %{_libdir}/nautilus/extensions-4/libnautilus-dropbox.so.bak ]; then
+    mv -f %{_libdir}/nautilus/extensions-4/libnautilus-dropbox.so{.bak,}
   fi
 fi
 
-rm -f %{_libdir}/nautilus/extensions-3.0/libnautilus-dropbox.so.bak
-rm -f %{_libdir}/nautilus/extensions-2.0/libnautilus-dropbox.so.bak
+rm -f %{_libdir}/nautilus/extensions-4/libnautilus-dropbox.so.bak
```
[the remark this answers is not in the document]
Comment (jewelpit) [reply generate-rpm.sh:228]: Oh, right. Old versions wouldn't be touching this path anyway.

**Comment by stevo-knievo** [comment]
I just upgraded to F37. But still can't find the package.

`dnf search nautilus-dropbox`

sorry if I missed something.

**Pull request merged**